### ISSUE TITLE
Couldn't resolve parser "apex-anonymous"

### ISSUE THREAD

**Issue by aallen-catalant** [dangmai/prettier-plugin-apex#712]
Hi all,

Long winded post here looking for some help as I've been banging my head against a wall here for a bit.

Trying to spin up a docker container to run this prettier plugin on my Apex files using the following dockerfile:

syntax=docker/dockerfile:1

FROM tmknom/prettier

USER root

RUN npm install --global prettier prettier-plugin-apex

RUN apk update && apk add bash

It is able to install into the node_modules directory, as I can find the plugin in the container shell using `find / -name node_modules`

Yet when I run `prettier --debug-check /path/to/class.cls` it will return back with one of two error messages.

The first one is when I have our .prettierrc file set up without an override parser option for .cls files, and that error message is `No parser could be inferred for file`

The second one is when I add in that override parser into that prettierrc file, `{
      "files": "**/lwc/**/*.html",
      "options": { "parser": "lwc" }
    },
    {
      "files": "*.cls",
      "options": { "parser": "apex-anonymous" }
    },
    {
      "files": "*.{cmp,page,component}",
      "options": { "parser": "html" }
    }`

and I receive`Couldn't resolve parser "apex-anonymous"`

The parser seems to be installing correctly, and its installed globally. I'm running these commands as root in the container so it shouldn't be a permissions issue. There must be something I am missing here, any help would be greatly appreciated. If this is not the right spot for this, also let me know and I can take it down. Thanks so much

**Comment by dangmai** [comment]
Hi! I am happy to help with this. Could you run these commands as part of the Dockerfile (after the `npm install` step) and copy/paste the results here?

```
npm ls --global prettier
npm ls --global prettier-plugin-apex
```

Thanks!

**Comment by aallen-catalant** [comment]
Thanks so much for the response! I was added in those two commands to my dockerfile and received the following:

#10 [3/6] RUN npm ls --global prettier
#10 sha256:ec8cfa9adafac03ca0932788a533e6a23ed04cfc8567b2bc168007d7733425ef
#10 1.413 /usr/local/lib
#10 1.413 +-- prettier-plugin-apex@1.12.0
#10 1.413 | `-- prettier@2.8.0 deduped
#10 1.413 `-- prettier@2.8.0
#10 1.413 
#10 DONE 1.4s

#11 [4/6] RUN npm ls --global prettier-plugin-apex
#11 sha256:b67f0f014c0b6d2e178d2d8626daa068e27f0647a9bdac2f40de039b7ab58623
#11 1.439 /usr/local/lib
#11 1.439 `-- prettier-plugin-apex@1.12.0
#11 1.439 
#11 DONE 1.5s


And when I ran `which prettier` I got this : /home/nonroot/node_modules/.bin/prettier


Thanks so much!

**Comment by dangmai** [comment]
Yeah so the issue is that there's a conflict of Prettier versions - the one you're running is embedded in the base image, _not_ the one that you installed. That version doesn't have access to Prettier Apex. If you're planning to install Prettier anyway, you can choose a different base image, for example `node:lts-alpine`

I'm going to close this ticket since it's not an issue with this library, but feel free to add more comments here if you need help.

**Comment by aallen-catalant** [comment]
@dangmai That was able to get me onto the right path, thanks. Issue was that base image, once I switched to alpine it was able to run (albeit with a different error). That different error was a parseTextWithSpawn error that I was able to fix after referencing a different issue here that told me to install Java into my container; is that a dependency I missed somewhere in the docs? If not, figured I'd let you know to add that note.
